bank_component.rb

```diff
@@ -0,0 +1,97 @@
+=begin
+ Feature 1: 
+ This feature will allow user to either sign up for an account.
+ Or log in with their last name,last 4 digits of their social security number, and password.        
+=end
+
+
+
+#class for log in and sign up method
+class LogInSignUp
+
+    @@user_info=[
+        {first:'andy', last:'park', password:'hello123', ssn:'123456789'},
+        {first:'Jason', last:'kim', password:'bye123', ssn:'135791098'}
+    ]
+
+    #Instance variable to determine if the user's credential is validated
+    @user_credential_validation = false
+    
+    attr_accessor :last_name, :password, :ssn
+
+    def initialize(last_name, password, ssn)
+        @last_name = last_name
+        @password=password
+        @ssn=ssn
+    end
+
+    def validate_log_in_credentials
+        @@user_info.each do |x|
+            if x[:last].to_s == @last_name && x[:password] == @password && x[:ssn][5,8] == @ssn
+                puts "Welcome back #{x[:first]} #{x[:last]}!"
+                return
+            else
+                puts "sorry access denied!"
+                return
+            end
+        end
+    end
+
+    def create_new_user_account
+      puts "HAHAH!"
+    end
+
+end
+
+
+#############################################
+######### Main program starts here!##########
+#############################################
+puts "Welcome to Bank of Invoca!"
+print "Are you an existing user?(Y/N):"
+$existing_new_user = gets.chomp.downcase
```

Comment:
Global vars are extremely rare to use and can be very problematic. Avoid at all costs! 🔴 
Class Variables are likewise very rare and worth avoiding. 🟠 
Instance Variables and class Constants are good 🟢